src/wp-includes/block-template-utils.php

```diff
@@ -724,6 +724,68 @@ function _wp_build_title_and_description_for_taxonomy_block_template( $taxonomy,
 	return true;
 }
 
+/**
+ * Build a block template object for based on a WP_Post object.
+ * 
+ * @since 6.5.0
+ * @access private
+ *
+ * @param WP_Post $post Template post.
+ * @param array $additional_fields Additional fields to add to the template object.
+ * @param bool $hook_blocks Whether to insert hooked blocks into the content or not.
+ * @return WP_Block_Template|WP_Error Template or error object.
+ */
+function _build_block_template_object_from_wp_post_object( $post, $additional_fields = array(), $hook_blocks = true ) {
+	$default_template_types = get_default_block_template_types();
+	$template_file  = _get_block_template_file( $post->post_type, $post->post_name );
+
+	$template = new WP_Block_Template();
+	$template->content 		  = $post->post_content;
+	$template->slug 		  = $post->post_name;
+	$template->type           = $post->post_type;
+	$template->description    = $post->post_excerpt;
+	$template->title          = $post->post_title;
+	$template->status         = $post->post_status;
+
+	// Set default values for properties expected to be set.
+	$template->wp_id 		  = null;
+	$template->modified 	  = null;
+	$template->is_custom      = null;
+	$template->has_theme_file = null;
+	$template->author		  = null;
+	$template->area		      = null;
+	$template->post_types	  = null;
+
+	if (
+		'wp_template' === $post->post_type &&
+		isset( $template_file['postTypes'] ) &&
+		isset( $additional_fields['has_theme_file'] ) &&
+		$additional_fields['has_theme_file']
+	) {
+		$template->post_types = $template_file['postTypes'];
+	}
+
+	if ( 'wp_template' === $post->post_type && isset( $default_template_types[ $template->slug ] ) ) {
+		$template->is_custom = false;
+	}
+
+	if ( is_array( $additional_fields ) && ! empty( $additional_fields ) ) {
+		foreach ( $additional_fields as $key => $value ) {
+			$template->{$key} = $value;
+		}
+	}
+
+	if ( $hook_blocks ) {
+		$hooked_blocks = get_hooked_blocks();
+		if ( ! empty( $hooked_blocks ) || has_filter( 'hooked_block_types' ) ) {
+			$before_block_visitor = make_before_block_visitor( $hooked_blocks, $template );
+			$after_block_visitor  = make_after_block_visitor( $hooked_blocks, $template );
+			$blocks               = parse_blocks( $template->content );
+			$template->content    = traverse_and_serialize_blocks( $blocks, $before_block_visitor, $after_block_visitor );
+		}
+	}
```

Comment:
I _think_ I'd remove the Block Hooks logic from `_build_block_template_object_from_wp_post_object` (and keep it one level higher, i.e. in `_build_block_template_result_from_post`).

(If we kept it here, wouldn't we also need to keep the filter override to suppress hooked blocks -- i.e. the `add_filter( 'hooked_block_types', '__return_empty_array', 99999, 0 );` stuff?)

Finally, I think it's fine to have a function that takes a `WP_Post` and creates a `WP_Block_Template` from it; if the caller also wants to inject hooked blocks, they can do that separately; after all, they'll have all the data they need at that point (contained in the `WP_Block_Template`).

Reply:
Yeah, I was merely experimenting with this and also concluded its better to live outside the function because the concerns are completely separate I feel.